src/ui-main.js

```diff
@@ -0,0 +1,181 @@
+/*package annotator.ui */
+"use strict";
+
+var adder = annotator.ui.adder;
+var editor = aca.ui.editor;
+var highlighter = annotator.ui.highlighter;
+var textselector = annotator.ui.textselector;
+var viewer = annotator.ui.viewer
+
+// Returns the absolute position of the mouse relative to the top-left rendered
+// corner of the page (taking into account padding/margin/border on the body
+// element as necessary).
+function mousePosition(event) {
+    var body = document.body;
+    var offset = {top: 0, left: 0};
+
+    if ($(body).css('position') !== "static") {
+        offset = $(body).offset();
+    }
+
+    return {
+        top: event.pageY - offset.top,
+        left: event.pageX - offset.left
+    };
+}
+
+// annotationFactory returns a function that can be used to construct an
+// annotation from a list of selected ranges.
+function annotationFactory(contextEl, ignoreSelector) {
+    return function (ranges) {
+        var text = [],
+            serializedRanges = [];
+
+        for (var i = 0, len = ranges.length; i < len; i++) {
+            var r = ranges[i];
+            text.push(String.prototype.trim(r.text()));
+            serializedRanges.push(r.serialize(contextEl, ignoreSelector));
+        }
+
+        return {
+            quote: text.join(' / '),
+            ranges: serializedRanges
+        };
+    };
+}
+
+/**
+ * function:: main([options])
+ *
+ * A module that provides a default user interface for Annotator that allows
+ * users to create annotations by selecting text within (a part of) the
+ * document.
+ *
+ * Example::
+ *
+ *     app.include(annotator.ui.main);
+ *
+ * :param Object options:
+ *
+ *   .. attribute:: options.element
+ *
+ *      A DOM element to which event listeners are bound. Defaults to
+ *      ``document.body``, allowing annotation of the whole document.
+ *
+ *   .. attribute:: options.editorExtensions
+ *
+ *      An array of editor extensions. See the
+ *      :class:`~annotator.ui.editor.Editor` documentation for details of editor
+ *      extensions.
+ *
+ *   .. attribute:: options.viewerExtensions
+ *
+ *      An array of viewer extensions. See the
+ *      :class:`~annotator.ui.viewer.Viewer` documentation for details of viewer
+ *      extensions.
+ *
+ */
+function main(options) {
+    if (typeof options === 'undefined' || options === null) {
+        options = {};
+    }
```

Comment:
`if (!options) { options = {}; } `
should be the same, or my favorite
`options || (options = {});`